contracts/zero-ex/contracts/src/vendor/v3/IExchange.sol

```diff
@@ -0,0 +1,107 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+
+/// @dev Interface to the V3 Exchange.
+interface IExchange {
```

Comment:
jw what's the etymology of this "vendor" directory?

Reply:
Vendor is convention for third party stuff. Right now it's just the exchange but in the future I would put uniswap, oasis, kyber interfaces in there. https://softwareengineering.stackexchange.com/questions/123305/what-is-the-difference-between-the-lib-and-vendor-folders